feat(exercises): :sparkles: Show a modal to select the exercises of an existing ExerciseList

samuelsainz opened the pull request:
## Description
<!--- Describe your changes in detail -->

In the View to see the details of an ExerciseList, we want to have a + button in the toolbar to add other exercises. 
To do that the SelectExercisesView is shown as a modal when the + button is tapped. 

### Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in a discussion first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
<!--- If this Pull Request closes an issue, please add `Closes #XXXX` -->

Closes #15 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Tested the next cases: 

Case 1 
* Create a new list 
* Navigate to the list
* Tap the + button in the toolbar 
* Select an exercise 
* Navigate back to the list 

**Result**: The exercise is shown in the list 

Case 2 
* Navigate to the list created in Case 1 
* Tap the + button in the toolbar
* Remove the exercise selected 
* Navigate back to the list 

**Result**: No exercises are shown 

 Case 3
* Navigate to the list created in Case 1 
* Tap the + button in the toolbar
* Select multiple exercises  
* Navigate back to the list 

**Result**: The correct exercises are shown 

Case 4: 
* Kill the app 
* Open the app again 
* Navigate to the list created in Case 1 

**Result**: The list created in Case 1 contains the right exercises  

## Screenshots (if appropriate):

![peakfit-screenrecording](https://github.com/user-attachments/assets/abd3d959-4f1c-4b14-8edc-325ed56a2519)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
<!--- Only one box should be checked -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer tooling (fix or improve developer tooling)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

heyjaywilson commented:
This is a good start. Only thing is can the plus button be added into the menu that already exists with a title "Add Exercise"?

samuelsainz commented:
@heyjaywilson sure thing! Sorry that I missed the new menu, I've rebased and added the option to the menu in another section.  

![Simulator Screenshot - iPhone 16 Pro - 2024-10-20 at 17 53 29](https://github.com/user-attachments/assets/8f536c40-a9e1-4f9e-8010-0dbcb53aaa2c)

let me know if that works :)
